test/LinalgTransform/expert.mlir

```diff
@@ -0,0 +1,89 @@
+// RUN: mlir-proto-opt -linalg-transform-expert-expansion %s | FileCheck %s --check-prefix=EXPAND
+// RUN: mlir-proto-opt -linalg-transform-expert-expansion -linalg-interp-transforms %s | FileCheck %s
+
+// CHECK-LABEL: func @matmul_tensors
+// CHECK-NOT: linalg
+// CHECK: llvm
+func @matmul_tensors(
+  %arg0: tensor<128x128xf32>, %arg1: tensor<128x128xf32>, %arg2: tensor<128x128xf32> { linalg.inplaceable = true})
+    -> tensor<128x128xf32> {
+  %0 = linalg.matmul  ins(%arg0, %arg1: tensor<128x128xf32>, tensor<128x128xf32>)
+                     outs(%arg2: tensor<128x128xf32>)
+    -> tensor<128x128xf32>
+
+  return %0 : tensor<128x128xf32>
+}
+
+pdl.pattern @pdl_target : benefit(1) {
+  %args = pdl.operands
+  %results = pdl.types
+  %0 = pdl.operation "linalg.matmul"(%args : !pdl.range<value>) -> (%results : !pdl.range<type>)
+  pdl.apply_native_constraint "nestedInFunc"[@matmul_tensors](%0 : !pdl.operation)
+  // TODO: we don't want this, but it is the required terminator for pdl.pattern
+  pdl.rewrite %0 with "linalg_transform.apply"
+}
+
+linalg_transform.sequence {
+  // This should match the strategy below.
+  // EXPAND-NOT: expert when
+  // EXPAND: %[[HANDLE:.*]] = tile when @pdl_target {sizes = [4, 4, 4]}
+  // EXPAND: %[[HANDLE2:.*]] = vectorize %[[HANDLE]] {vectorize_padding = true}
+  // EXPAND: bufferize
+  // EXPAND: lower_vectors {multireduction_lowering = "innerreduce"}
+  // EXPAND: lower_to_llvm
+  expert when @pdl_target "single_tiling"
+  {
+    tile_sizes = [4, 4, 4],
+    vectorize_padding = true,
+    multireduction_lowering = "innerreduce"
+  }
+}
+
+// CHECK-NOT: @strategies
+// EXPAND-NOT: @strategies
+module @strategies {
+  pdl.pattern @single_tiling : benefit(1) {
+    %tile_sizes = pdl.attribute
+    %vectorize_padding = pdl.attribute
+    %multireduction_lowering = pdl.attribute
+    %name = pdl.attribute : "single_tiling"
+    %target = pdl.attribute
+    %transformed = pdl.type : !pdl.operation
```

Comment:
Specifying `!pdl.operation` is redundant because the op always returns an operation.